enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+be limited (for example, by running a single API service).
+
+Failures in edge deployments are frequently resolved by re-imaging or
+physically replacing the entire host. Combining this fact with the
+previous observation about the approach to providing highly-available
+services lets us draw the conclusion that in-place upgrades do not
+need to be supported by the first iteration of this cluster profile.
+
+The cluster profile will be sufficiently generic to cater to a variety
+of edge computing use cases. As such, OpenShift's usual cluster
+configuration mechanisms will be favored where there is a likelihood
+that there will be edge computing use case with differing requirements.
+For example, there is no expectation that this new profile will use
+the real-time kernel by default - this will continue to be a 
+`MachineConfig` choice as per
+[enhancements/support-for-realtime-kernel](https://github.com/openshift/enhancements/blob/master/enhancements/support-for-realtime-kernel.md). *See open questions*
+
+### User Stories
+
+#### As a user, I can deploy OpenShift in a supported single-node configuration
+
+A user will be able to run the OpenShift installer to create a single-node
+deployment, with some limitations (see non-goals above). The user
+will not require special support exceptions to receive technical assistance
+for the features supported by the configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Some OpenShift components (such as Etcd and Ingress) require
+a minimum of 2 or 3 nodes. The `single-node-production-edge`
+cluster profile will configure these components as appropriate
+for a single node.
+
+Known gaps:
+- `cluster-etcd-operator` will not deploy the etcd cluster without minimum of 3 master nodes (can be changed by enabling `useUnsupportedUnsafeNonHANonProductionUnstableEtcd`)
+- Even with the unsupported feature flag, `etcd-quorum-guard` still requires 3 nodes due to its replica count.
+- `cluster-authentication-operator` will not deploy `OAuthServer` without minimum of 3 master nodes (can be change by enabling `useUnsupportedUnsafeNonHANonProductionUnstableOAuthServer`)
+- `cluster-ingress-operator` deploys the router with 2 replicas. On a single node one will fail to start and the ingress will show as degraded.
+
+When we are deploying a cluster with the `single-node-production-edge`
+cluster profile the relevant operators should support a non-HA
+configuration that makes the correct adjustments to the deployment
+(e.g., `cluster-ingress-operator` should deploy a single router,
+`cluster-etcd-operator` should deploy the `etcd-member`
+[without waiting for 3 master nodes](https://github.com/openshift/cluster-etcd-operator/blob/98590e6ecfe282735c4eff01432ae40b29f81202/pkg/etcdenvvar/etcd_env.go#L72))
+
+In addition, some components are not relevant for this cluster
+profile (e.g. console, cluster-autoscaler, marketplace?) and shouldn't
```

Comment:
marketplace/OLM is necessary. Both Performance Addon Operator and sriov operators are deployed that way.

Reply:
You need an installation that understands OLM's API surface, that does not necessarily mean you require OLM running actively on the machine at all times. I could see an alternative where OLM's controllers are run in a one-shot install / upgrade mode as opposed to constantly running. It may come down to where resource utilization needs to be cut.

If upgrades initially will happen by reimaging the machine, then is there a reason to have OLM continuously running and checking its catalog for updates? I think this depends on whether there are likely to be updates to the optional components independent of reimaging this single node cluster. The other purpose OLM serves if it is running is to provide general metrics about the optional components.

Reply:
Ok, I think I can accept that. We need a way to install the operators, but there does not have to be a constant updates checking loop.